lib/floe/workflow/state.rb

```diff
@@ -4,32 +4,31 @@ module Floe
   class Workflow
     class State
       include Logging
+      include ValidationMixin
 
       class << self
-        def build!(workflow, name, payload)
+        def build!(workflow, full_name, payload)
           state_type = payload["Type"]
-          raise Floe::InvalidWorkflowError, "Missing \"Type\" field in state [#{name}]" if payload["Type"].nil?
+          parser_missing_field!(full_name, "Type") if payload["Type"].nil?
```

Comment:
Just naming stuff, but wonder if validate_missing_field! would be better?

Reply:
This is not checking values but rather throwing an error.
It is essentially a wrapper that generates the proper text.

(I would prefer if all/most errors were of the form "$State field $Field $message", but had trouble keeping this language active vs passive, so I punted to `statelint` wording, and kept this as a place holder to fix.)

I will look for a better phrase

Reply:
going to change to `missing_field_error!`